### PR TITLE
os/filestore: when print log, use __func__ instead of hard code function name

### PR DESCRIPTION
**Pull request by mychoxin**
when print log, use __func__ instead of hard code function name

Signed-off-by: mychoxin <mychoxin@gmail.com>

### CODE REVIEW

**Comment by mychoxin** [comment]
@liupan1111 @tchaicov

File: src/os/filestore/FileStore.cc [hunk 1]
```diff
@@ -1742,17 +1742,17 @@ int FileStore::mount()
   if (!m_disable_wbthrottle) {
     wbthrottle.start();
   } else {
-    dout(0) << "mount INFO: WbThrottle is disabled" << dendl;
+    dout(0) << __func__ << " INFO: WbThrottle is disabled" << dendl;
     if (cct->_conf->filestore_odsync_write) {
-      dout(0) << "mount INFO: O_DSYNC write is enabled" << dendl;
+      dout(0) << __func__ << " O_DSYNC write is enabled" << dendl;
```
Comment (tchaikov) [inline src/os/filestore/FileStore.cc]: maybe we can keep `INFO` to be backward compatible, unless we have a good reason not to do so?

File: src/os/filestore/FileStore.cc [hunk 2]
```diff
@@ -1866,7 +1866,7 @@ void FileStore::init_temp_collections()
 
 int FileStore::umount()
 {
-  dout(5) << "umount " << basedir << dendl;
+  dout(5) << __func__ << ": " << basedir << dendl;
```
Comment (tchaikov) [inline src/os/filestore/FileStore.cc]: why are you adding a colon here but not adding it otherwhere?

**Comment by tchaikov** [comment]
@mychoxin also, please note `__func__` is not a macro. it is a constant.

File: src/os/filestore/FileStore.cc (hunk 2 above)
Comment (mychoxin) [inline src/os/filestore/FileStore.cc]: some where there is colon, and other where there is no colon, so I should add colon at all place to keep the same.

**Comment by mychoxin** [comment]
@tchaikov already modified, please take a look.

**Comment by mychoxin** [comment]
and @joscollin, please take a look too.

**Comment by wjwithagen** [comment]
While at it...
Perhaps you could even change this into a macro that looks like:
```
#define __FUNC__  __func__ << "(" << __LINE << ")" 
```

Which makes inpointing the log even easier, especially during debugging.
Adn if you'd like to rewrite the output, you need to do that only at one location.

But I'm not sure when __func__ and __LINE__ are evaluated. Could be that the preprocessor already takes care of __LINE__. and you end up the the linenumber of the macro.
Then you have to start parameterizing it.

**Comment by joscollin** [comment]
@wjwithagen I was going to approve this, but suddenly saw your comment. So Could you please do "Request Changes" ?

**Comment by mychoxin** [comment]
@wjwithagen good idea

**Comment by mychoxin** [comment]
@wjwithagen 
I tested __LINE__, it works fine(the line number is file line number, not offset function).
and my format will be like: "function:handle_remove_dev, line:343", 
because it seems that the format like: "handle_remove_dev(343)" does not be clear enough.

**Comment by mychoxin** [comment]
@joscollin @wjwithagen ping
any other suggestion?

**Comment by joscollin** [comment]
@mychoxin Your PR contains 2 commits, one from another user. You need a rebase.

**Comment by mychoxin** [comment]
@joscollin @wjwithagen rebase done

File: src/os/filestore/FileStore.cc [hunk 3]
```diff
@@ -1332,7 +1333,7 @@ int FileStore::read_op_seq(uint64_t *seq)
   memset(s, 0, sizeof(s));
   int ret = safe_read(op_fd, s, sizeof(s) - 1);
   if (ret < 0) {
-    derr << "error reading " << current_op_seq_fn << ": " << cpp_strerror(ret) << dendl;
+    derr << __FUNC__ << "error reading " << current_op_seq_fn << ": " << cpp_strerror(ret) << dendl;
```
Comment (joscollin) [inline src/os/filestore/FileStore.cc]: Missed colon

File: src/os/filestore/FileStore.cc [hunk 4]
```diff
@@ -3165,12 +3166,12 @@ int FileStore::read(
...
   const coll_t& cid = !_need_temp_object_collection(_cid, oid) ? _cid : _cid.get_temp();
 
-  dout(15) << "read " << cid << "/" << oid << " " << offset << "~" << len << dendl;
+  dout(15) << __FUNC__ << ": " << cid << "/" << oid << " " << offset << "~" << len << dendl;
 
   FDRef fd;
   int r = lfn_open(cid, oid, false, &fd);
   if (r < 0) {
-    dout(10) << "FileStore::read(" << cid << "/" << oid << ") open error: "
+    dout(10) << __FUNC__ << "(" << cid << "/" << oid << ") open error: "
```
Comment (joscollin) [inline src/os/filestore/FileStore.cc]: Missed colon or is it intentional  ?

File: src/os/filestore/FileStore.cc [hunk 5]
```diff
@@ -3267,7 +3268,7 @@ int FileStore::_do_fiemap(int fd, uint64_t offset, size_t len,
   while (i < fiemap->fm_mapped_extents) {
     struct fiemap_extent *next = extent + 1;
 
-    dout(10) << "FileStore::fiemap() fm_mapped_extents=" << fiemap->fm_mapped_extents
+    dout(10) << __FUNC__ << "() fm_mapped_extents=" << fiemap->fm_mapped_extents
```
Comment (joscollin) [inline src/os/filestore/FileStore.cc]: Check the format here.

File: src/os/filestore/FileStore.cc [hunk 6]
```diff
@@ -107,6 +107,7 @@ using ceph::crypto::SHA1;
 #define XATTR_SPILL_OUT_NAME "user.cephos.spill_out"
 #define XATTR_NO_SPILL_OUT "0"
 #define XATTR_SPILL_OUT "1"
+#define __FUNC__ "function:" << __func__ << ", line:" << __LINE__
```
Comment (wjwithagen) [inline src/os/filestore/FileStore.cc]: @mychoxin 
I would not make the __FUNC__ string too long. aan somethings like:
`func_so_and_so(12345)` would already be informative.
The other thing you might want to check, is whether `__func__` actually prints the function name, or that is also has the class prefix. (I think it does not have the prefix)

**Comment by mychoxin** [comment]
@wjwithagen @joscollin 
\_\_func\_\_ has no class prefix, but \_\_PRETTY\_FUNCTION\_\_is something good that we can use, here is my examples:
for a class member function, it prints like:
int head_info_parser::parse_info()
for a global function, it prints like: 
int handle_remove_dev(void*, dev_info*)
it includes return type, parameter type lists and class name if there exists.

**Comment by mychoxin** [comment]
@wjwithagen for default we use \_\_func\_\_, and if someone wants to debug and needs the complete function name then he can open the comment to use   \_\_PRETTY\_FUNCTION\_\_

File: src/os/filestore/FileStore.cc (hunk 4 above)
Comment (mychoxin) [inline src/os/filestore/FileStore.cc]: missed

**Comment by mychoxin** [comment]
@joscollin @wjwithagen ping

File: src/os/filestore/FileStore.cc [hunk 7]
```diff
@@ -3193,10 +3195,10 @@ int FileStore::read(
   bufferptr bptr(len);  // prealloc space for entire read
   got = safe_pread(**fd, bptr.c_str(), len, offset);
   if (got < 0) {
-    dout(10) << "FileStore::read(" << cid << "/" << oid << ") pread error: " << cpp_strerror(got) << dendl;
+    dout(10) << __FUNC__ << "(" << cid << "/" << oid << ") pread error: " << cpp_strerror(got) << dendl;
```
Comment (joscollin) [inline src/os/filestore/FileStore.cc]: Missed colon or is it intentional ?

File: src/os/filestore/FileStore.cc [hunk 8]
```diff
@@ -3193,10 +3195,10 @@ int FileStore::read(
...
     lfn_close(fd);
     if (!(allow_eio || !m_filestore_fail_eio || got != -EIO)) {
-      derr << "FileStore::read(" << cid << "/" << oid << ") pread error: " << cpp_strerror(got) << dendl;
+      derr << __FUNC__ << "(" << cid << "/" << oid << ") pread error: " << cpp_strerror(got) << dendl;
```
Comment (joscollin) [inline src/os/filestore/FileStore.cc]: Missed colon or is it intentional ?

File: src/os/filestore/FileStore.cc (hunk 7 above)
Comment (mychoxin) [inline src/os/filestore/FileStore.cc]: missed colon, i think the parenthesis do not like parameter list

File: src/os/filestore/FileStore.cc (hunk 8 above)
Comment (mychoxin) [inline src/os/filestore/FileStore.cc]: missed colon, i think the parenthesis do not like parameter list

**Comment by mychoxin** [comment]
@joscollin @wjwithagen please take a look.

**Comment by mychoxin** [comment]
@wjwithagen ping

File: src/os/filestore/FileStore.cc (hunk 6 above)
Comment (mychoxin) [inline src/os/filestore/FileStore.cc]: @wjwithagen \_\_func\_\_ has no class prefix, and there is no other way to just get className::funName,  \_\_PRETTY_FUNCTION\_\_ is too informative, may be i can trim from \_\_PRETTY_FUNCTION\_\_, so what's your opinion?

**Comment by liupan1111** [comment]
@wjwithagen ping?

File: src/os/filestore/FileStore.cc [hunk 9]
```diff
@@ -107,6 +107,8 @@ using ceph::crypto::SHA1;
 #define XATTR_SPILL_OUT_NAME "user.cephos.spill_out"
 #define XATTR_NO_SPILL_OUT "0"
 #define XATTR_SPILL_OUT "1"
+#define __FUNC__ __func__ << "(" << __LINE__ << ")"
+//#define __FUNC__ __PRETTY_FUNCTION__ << "(" << __LINE__ << ")"
```
Comment (joscollin) [inline src/os/filestore/FileStore.cc]: Could you please remove the commented code ? This was not there when I approved the changes !!!

Comment (mychoxin) [inline src/os/filestore/FileStore.cc]: I removed it.

**Comment by joscollin** [comment]
Jenkins Retest this please

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-21_22:08:07-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**